client/src/components/Query/Statement.jsx

```diff
@@ -331,20 +334,55 @@ class Statement extends React.Component {
   }
 
   handleRemoveChecked() {
+
     if (this.isRemovable()) {
       const { checkedQueries, draft } = this.state;
-      this.commit(draft);
-      const keysToRemove = checkedQueries.reduce((accumulator, key) => [...accumulator, { key }], []);
-      pullAllBy(draft, keysToRemove, 'key');
-      this.setState({
-        draft,
-        checkedQueries: [],
-        queriesChecksAreIndeterminate: false,
-        queriesAreAllChecked: false,
-      });
+
+      let keys = []
+      for(let d of draft){
+           d.instructions.map(i => {if(i.type === "subquery"){
+                                        checkedQueries.map(c => c === i.data.query ? keys.push({"key" :d.key}) : null)
+                                   }})
+      }
+      keys.length != 0 ? this.showDeleteConfirm(keys) : this.confirmRemove()
+
     }
   }
 
+
+  showDeleteConfirm(delSubQuery) {
+      const { confirm } = Modal;
+      const that =this
+        confirm({
+          title: 'Are you sure you want to delete this query?',
```

Comment:
@claudia1296 Could we add these strings to the locales so that we can translate them using `intl`?